### DISCUSSION TITLE
Allays

### DISCUSSION THREAD

**Discussion by object-Object** [HexxyCraft/modpack#106] [General]
There are no allays in the world. How should we make them available?

Ideas from me and Llama:
* Do an event of some sort where we spawn / hand out allays
* Make caged allays spawn in some loot chests (eg. pillager outposts)
* Add caged allays to evoker drops (available via raids)

**Comment by FractalineHue** [comment]
Caged allays from evoker drops sounds like the best to me?

**Comment by walksanatora** [comment]
~~craftable allays~~

**Comment by JohnDog3112** [comment]
I don’t know if it’s worth adding a mod for it, but carpet sky additions adds a mini game where playing the right sequence of notes to a vex converts it to an allay. [https://modrinth.com/mod/carpet-sky-additions/changelog](https://modrinth.com/mod/carpet-sky-additions/changelog)

**Reply by LlamaSageR** [reply]
The other changes are interesting. I'll look into the mod.

**Comment by computer-whisperer** [comment]
Make allay spawning require some obscure and/r difficult-to-arrange interaction. Maybe one of the following:

- Have the wither kill the ender dragon
- Toss a phial of exactly 11111 amethyst dust into lava
- Have the warden die of suffocation damage (not sure if possible)
- Mass mindflay variant

Something similar would also work for elytra renew-ability

**Reply by JohnDog3112** [reply]
For mass mind flay, maybe make it so you can cage a vex and you mass mind flay something like golems, some tamed animal, or a somewhat related villager progression on it.

**Reply by walksanatora** [reply]
"fun" fact: you can include non-villagers in Mass-Mindflay and do things with them (only the villagers/raiders get mindflayed)

**Reply by walksanatora** [reply]
so if you *want* I could add that (either as a seperate mod or figure out of CraftTweaker can do it)

**Reply by walksanatora** [reply]
Adding CraftTweaker support to HexTweaks (note: may make it so you you can access more hexcasting stuff... *custom spells from ZenScript anyone*)

**Reply by object-Object** [reply]
Like custom recipes? CT fully supports adding stuff to tags etc, and custom recipes can be done with ct's json recipe utils afaik.

**Reply by walksanatora** [reply]
Custom Mass Mindflay recipes

**Reply by walksanatora** [reply]
(they are actually java functions not json recipes, although I *should* look into setting up json recipes)

**Reply by object-Object** [reply]
that would probably be ideal - I think brainsweep does that, you could use that as a base. either way, CT support would definitely be *nice* so it's somewhat more ergonomic than writing raw json within zs

**Reply by walksanatora** [reply]
(procedes to write the most *bespoke* CT api known to mankind)

**Reply by computer-whisperer** [reply]
mass mind flay 9 cats into a vex?

**Reply by LlamaSageR** [reply]
https://modrinth.com/mod/carpet-sky-additions adds an interesting way to create Allays, namely playing a specific noteblock song near a Vex. As this is a Carpet mod, it's also configurable.

**Comment by maraxin** [comment]
Yo I'm here really really late, but how about caged allays being in dungeon chests? This is still challenge, while everyone can get allays

**Reply by LlamaSageR** [reply]
I feel that using Dungeon Loot is less ideal, since it already has a lot of items in it. Having a dedicated method of procuring Allays, especially one that incentivizes a mechanic that players don't really interact with (Raids), would be better.

**Reply by walksanatora** [reply]
dungeon chest could be easily automated via loonium so this makes them *very* common once a loonium setup is complete
